substra/sdk/backends/remote/backend.py

```diff
@@ -109,24 +111,155 @@ def add(self, spec, exist_ok=False, spec_options=None):
                 exist_ok=exist_ok,
                 spec_options=spec_options,
             )
+        elif asset_type == schemas.Type.ComputePlan:
+            # Compute plan auto batching feature
+            return self.__auto_batching_compute_plan(
+                spec=spec,
+                exist_ok=exist_ok,
+                spec_options=spec_options,
+            )
 
         with spec.build_request_kwargs(**spec_options) as (data, files):
             response = self._add(asset_type, data, files=files, exist_ok=exist_ok)
         # The backend has inconsistent API responses when getting or adding an asset
         # (with much less data when responding to adds).
         # A second GET request hides the discrepancies.
-        if asset_type == schemas.Type.ComputePlan:
-            return response
-
         key = _get_asset_key(response)
         return self.get(asset_type, key)
 
+    def __fill_tuple_lists(
+        self,
+        spec,
+        tuple_graph,
+        traintuples,
+        aggregatetuples,
+        composite_traintuples,
+        testtuples_by_train_id,
+    ):
+        for elem_id, _ in tuple_graph:
+            if elem_id in traintuples:
+                spec.traintuples.append(traintuples[elem_id])
+            elif elem_id in aggregatetuples:
+                spec.aggregatetuples.append(aggregatetuples[elem_id])
+            elif elem_id in composite_traintuples:
+                spec.composite_traintuples.append(composite_traintuples[elem_id])
+            else:
+                raise Exception("ERROR")
+            if elem_id in testtuples_by_train_id:
+                spec.testtuples.append(testtuples_by_train_id[elem_id])
+
+    def __get_testtuples_by_train_id(self, spec):
+        testtuples_by_train_id = dict()
+        for testtuple in spec.testtuples:
+            testtuples_by_train_id[testtuple.traintuple_id] = testtuple
+        return testtuples_by_train_id
+
+    def __auto_batching_compute_plan(self,
+                                     spec,
+                                     compute_plan_id=None,
+                                     exist_ok=None,
+                                     spec_options=None):
+        # Auto batching of the compute plan tuples
+
+        (
+            tuple_graph,
+            traintuples,
+            aggregatetuples,
+            composite_traintuples
+        ) = spec.get_dependency_graph()
+
+        compute_plan = None
+        visited = dict()
+        existing_keys = dict()
+        if compute_plan_id is not None:
```

Comment:
We could use `if compute_plan_id:` here, it would be a bit broader to trigger and easier to read.